src/components/profile.js

```diff
@@ -32,38 +37,86 @@ class Profile extends Component {
 		return view;
 	}
 
+	setChanges( key ) {
+		return value => {
+			let changes = {};
+			changes[ key ] = value;
+
+			this.setState( changes );
+
+			return this.state.profile.save(
+				changes,
+				{ patch: true }
+			);
```

Comment:
I agree this should be optimized and I want to abstract any interaction with the models and collections through src/requests, as fetching, changing, etc.